### PR TITLE
Tic Tac Toe game in javascript

### PR DESCRIPTION
**Pull request by ampaire**
We implemented the TIc tac toe game in javascript and these are the features of the game
- A user is able to feed in their names
- If a user does not fill in their names, by default, the game gives the names **Player1** and **player2**
- Players are taken to the board interface when they click on play
- Players take turns to print either **X** or **O** on the board
- Players can see who wins a game and a congratulatory message is sent out to them
- Players can also see if its a draw that is if there are no winning combinations
- Players can decide to play again or quit the game

### CODE REVIEW

File: js/dom.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/* eslint-disable func-names */
+// eslint-disable-next-line func-names
+// eslint-disable-next-line no-unused-vars
+const dom = (function () {
+  function getElement(id) {
```
Comment (bellom) [inline js/dom.js:5]: [Optional]
It would be great to practice `ES6` by using arrow function.

File: js/logic.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+  const id = dom.getId(_index);
+  if (engine.board[id] === '') {
+    engine.board[id] = sign;
+    dom.render(id, engine.board[id]);
+    return true;
+  }
+  return false;
+};
+
+let player1 = player('');
```
Comment (bellom) [inline js/logic.js]: Geat work using module/factory.... 👍 
But in this project we don't want to use global variable;
image

File: js/logic.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
+/* eslint-disable no-undef */
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-unused-expressions */
+// eslint-disable-next-line func-names
+const game = (function () {
```
Comment (bellom) [inline js/logic.js]: You have done well 👍 , You did awesome keeping the DOM from the logic.
Now, let follow the instructions from the Odin;
image
https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript

In another word; You need to create 3 separate objects, don't forget, no global variable.
1. Gameboard: uses an array to store the 'moves'
2. Player: players will be store as an object, which you already did.
3. Game: control the overflow of the game.


Example:
```
const Player = (x, y) => {
  return { x, y };
};
```
```
const GameBoard = (() => {
  let gameBoard = [ ......... ];
  return { gameBoard };
})();
```

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi Team :clap: 

### Awesome Design Guys!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**

**Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....**

**Best,**

**Oluwaseun J. Bello**
**_Slack Handle - @sean bellom_**

File: js/dom.js (hunk 1 above)
Comment (Tresor11) [reply js/dom.js:5]: Thank you for the suggestion we've used it now

File: js/logic.js (hunk 3 above)
Comment (Tresor11) [reply js/logic.js]: thank you for the suggestion we've created a player module to contain all our players

**Review by shloch — APPROVED** [review]
## perfect execution @ampaire , @Tresor11  for implementing changes

Screen Shot 2020-03-19 at 12 16 07 AM
accomplished

**Pull request merged**